source/common/formatter/substitution_formatter.cc

```diff
@@ -1959,11 +1959,12 @@ MetadataFormatter::formatMetadata(const envoy::config::core::v3::Metadata& metad
     str = value.string_value();
   } else {
 #ifdef ENVOY_ENABLE_YAML
-    ProtobufUtil::StatusOr<std::string> json_or_error =
+    absl::StatusOr<std::string> json_or_error =
         MessageUtil::getJsonStringFromMessage(value, false, true);
-    ENVOY_BUG(json_or_error.ok(), "Failed to parse json");
     if (json_or_error.ok()) {
       str = json_or_error.value();
+    } else {
+      str = json_or_error.status().message();
```

Comment:
This may be a user-facing change.
Is it required, or can it be done in a followup PR (maybe w/ a runtime flag)?

Reply:
Not, unless we fork the previous implementation of the protobuf JSON formatter. I have discussed this on the maintainers Slack channel and the feedback was that this is edge case that does not warrant forking and maintaining old protobuf code.

Reply:
SGTM, thanks!